rpcserver.go

```diff
@@ -5233,3 +5241,37 @@ func (r *rpcServer) ChannelAcceptor(stream lnrpc.Lightning_ChannelAcceptorServer
 		}
 	}
 }
+
+// NewMacaroon allows the creation of a new macaroon with custom read and write
+// permissions. No first-party caveats are added since this can be done offline.
+func (r *rpcServer) NewMacaroon(ctx context.Context,
+	req *lnrpc.NewMacaroonRequest) (*lnrpc.NewMacaroonResponse, error) {
+
+	rpcsLog.Debugf("[newmacaroon]")
+
+	// Map permission struct used by gRPC to the one used by the bakery.
+	requestedPermissions := make([]bakery.Op, len(req.Permissions))
+	for idx, permission := range req.Permissions {
+		requestedPermissions[idx] = bakery.Op{
+			Entity: permission.Entity,
+			Action: permission.Action,
+		}
+	}
+
+	// Bake new macaroon with the given permissions and send it binary
+	// serialized and hex encoded to the client.
+	newMac, err := r.server.macService.Oven.NewMacaroon(
```

Comment:
If you'd be following up this PR with more granular macaroon baking on the individual call level, how would that map to `bakery.Op`s?

Reply:
My idea for the follow-up PR would be that we would use a constant for the `entity` (for example `"rpc"`) and the full RPC URL (as seen in `rpcserver.go`, for example `"/lnrpc.Lightning/SendCoins"`) as the `action`. This would result in quite large macaroons but would offer most flexibility.

Reply:
Have we already heard of a use case where our partitioning is insufficient and this fine granularity is required?